xds/internal/client/client.go

```diff
@@ -195,7 +195,15 @@ type UpdateMetadata struct {
 type ListenerUpdate struct {
 	// RouteConfigName is the route configuration name corresponding to the
 	// target which is being watched through LDS.
+	//
+	// Only one of RouteConfigName and InlineRouteConfig is set.
 	RouteConfigName string
+	// InlineRouteConfig is the inline route configuration (RDS response)
+	// returned inside LDS.
+	//
+	// Only one of RouteConfigName and InlineRouteConfig is set.
+	InlineRouteConfig RouteConfigUpdate
```

Comment:
Is it required/necessary to be able to differentiate between an empty inline `RouteConfigUpdate` and a non-existent one?

Like, should we reject the resource if it does not contain both `RouteConfigName` and inline `RouteConfigUpdate`?

So, should we make this a pointer?

Reply:
Made this a pointer.

I was check if `RDS-name` is an empty string. But I think an explicit check for pointer is better.